src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.spec.ts

```diff
@@ -80,6 +80,12 @@ describe('TelemetryComponent', () => {
       expect(component).toBeTruthy();
     });
 
+    it('should show ident fields when checked ', () => {
```

Comment:
@nizamial09 @aaryanporwal This test is not helpful as it's not checking if the fields are hidden/shown on checking/unchecking, it's testing that the flag variable is updated accordingly, but the only way to guarantee that the fields are shown/hidden is to effectively check the visibility.
This is my suggestion:
```
diff --git a/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.html b/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.html
index 3aa55ddf9f8..e3c0a6c1739 100644
--- a/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.html
+++ b/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.html
@@ -127,7 +127,7 @@
                          class="custom-control-input"
                          id="channel_ident"
                          formControlName="channel_ident"
-                         (click)="toggleIndent()">
+                         (click)="toggleIdent()">
                   <label class="custom-control-label"
                          for="channel_ident"></label>
                 </div>
diff --git a/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.spec.ts b/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.spec.ts
index f60196f701e..ac3a44d215c 100644
--- a/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.spec.ts
+++ b/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.spec.ts
@@ -80,10 +80,27 @@ describe('TelemetryComponent', () => {
       expect(component).toBeTruthy();
     });
 
-    it('should show ident fields when checked ', () => {
-      component.showContactInfo = false;
-      component.toggleIndent();
-      expect(component.showContactInfo).toBe(true);
+    it('should show/hide ident fields when checking/unchecking', () => {
+      const getContactField = () =>
+        fixture.debugElement.nativeElement.querySelector('input[id=contact]');
+      const getDescriptionField = () =>
+        fixture.debugElement.nativeElement.querySelector('input[id=description]');
+
+      // Initially hidden.
+      expect(getContactField()).toBeFalsy();
+      expect(getDescriptionField()).toBeFalsy();
+
+      // Show fields.
+      component.toggleIdent();
+      fixture.detectChanges();
+      expect(getContactField()).toBeTruthy();
+      expect(getDescriptionField()).toBeTruthy();
+
+      // Hide fields.
+      component.toggleIdent();
+      fixture.detectChanges();
+      expect(getContactField()).toBeFalsy();
+      expect(getDescriptionField()).toBeFalsy();
     });
 
     it('should set module enability to true correctly', () => {
diff --git a/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.ts b/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.ts
index 777c38053ad..758612dd472 100644
--- a/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.ts
+++ b/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.ts
@@ -138,7 +138,7 @@ export class TelemetryComponent extends CdForm implements OnInit {
     );
   }
 
-  toggleIndent() {
+  toggleIdent() {
     this.showContactInfo = !this.showContactInfo;
   }
```

Reply:
Thanks! fixed in ad5b3f200529fc0bc511ce99eed338afcaef6a62